Source/WarpX.cpp

```diff
@@ -1871,6 +1876,20 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     current_fp[lev][1] = std::make_unique<MultiFab>(amrex::convert(ba,jy_nodal_flag),dm,ncomps,ngJ,tag("current_fp[y]"));
     current_fp[lev][2] = std::make_unique<MultiFab>(amrex::convert(ba,jz_nodal_flag),dm,ncomps,ngJ,tag("current_fp[z]"));
 
+    // Same as Bfield_fp/Efield_fp for reading external field data
+    ParmParse pp_warpx("warpx");
+    pp_warpx.query("B_ext_grid_init_style", WarpX::B_ext_grid_s);
+    pp_warpx.query("E_ext_grid_init_style", WarpX::E_ext_grid_s);
+    if ((WarpX::B_ext_grid_s=="read_from_file" || WarpX::E_ext_grid_s=="read_from_file") && lev==0)
+    {
+        Bfield_fp_external[lev][0] = std::make_unique<MultiFab>(amrex::convert(ba,Bx_nodal_flag),dm,ncomps,ngEB,tag("Bfield_fp_external[0]"));
+        Bfield_fp_external[lev][1] = std::make_unique<MultiFab>(amrex::convert(ba,By_nodal_flag),dm,ncomps,ngEB,tag("Bfield_fp_external[1]"));
+        Bfield_fp_external[lev][2] = std::make_unique<MultiFab>(amrex::convert(ba,Bz_nodal_flag),dm,ncomps,ngEB,tag("Bfield_fp_external[2]"));
+        Efield_fp_external[lev][0] = std::make_unique<MultiFab>(amrex::convert(ba,Ex_nodal_flag),dm,ncomps,ngEB,tag("Efield_fp_external[0]"));
+        Efield_fp_external[lev][1] = std::make_unique<MultiFab>(amrex::convert(ba,Ey_nodal_flag),dm,ncomps,ngEB,tag("Efield_fp_external[1]"));
+        Efield_fp_external[lev][2] = std::make_unique<MultiFab>(amrex::convert(ba,Ez_nodal_flag),dm,ncomps,ngEB,tag("Efield_fp_external[2]"));
+    }
```

Comment:
E and B fields should probably be handled separately, since just one or the other is a totally reasonable use case.